frontstage/views/passwords/forgot_password.py

```diff
@@ -45,15 +51,22 @@ def post_forgot_password():
             if exc.status_code == 404:
                 logger.error('Requesting password change for email registered'
                              ' on OAuth2 server but not in party service')
-                return render_template('passwords/forgot-password.check-email.html', email=email)
+                return redirect(url_for('passwords_bp.forgot_password_check_email', email=encoded_email))
             raise exc
 
         logger.info('Successfully sent password change request email')
-        return redirect(url_for('passwords_bp.forgot_password_check_email'))
+        return redirect(url_for('passwords_bp.forgot_password_check_email', email=encoded_email))
 
     return render_template('passwords/forgot-password.html', form=form, email=email)
 
 
 @passwords_bp.route('/forgot-password/check-email', methods=['GET'])
 def forgot_password_check_email():
-    return render_template('passwords/forgot-password.check-email.html')
+    encoded_email = request.args.get('email', None)
+
+    try:
+        email = s.loads(encoded_email)
+    except BadSignature:
+        raise InvalidURLSignature('Unable to decode email from URL')
```

Comment:
I don't think we need to raise a custom exception, we could just deal with the error here.